torch_geometric/nn/aggr/adaptive.py

```diff
@@ -0,0 +1,400 @@
+    . DeepSetsAggregation is a permutation-invariant operator.
+
+    Args:
+        max_num_nodes_in_dataset (int): Maximum number of nodes in a graph in
+            the dataset.
+        in_channels (int): Size of each input sample.
+        out_channels (int): Size of each output sample.
+        num_hidden_layers_phi (int): Number of hidden layers for the
+            :math:`\phi` MLP.
+        num_hidden_layers_rho (int): Number of hidden layers for the
+            :math:`\rho` MLP.
+        hidden_dim (int): Dimensions used for the hidden layers of the MLPs.
+
+    """
+    def __init__(self, max_num_nodes_in_dataset: int, in_channels: int,
+                 out_channels: int, num_hidden_layers_phi: int,
+                 num_hidden_layers_rho: int, hidden_dim: int):
+        super().__init__()
+
+        assert num_hidden_layers_phi >= 0, (
+            'Number of phi hidden layers must be non-negative.')
+        assert num_hidden_layers_rho >= 0, (
+            'Number of rho hidden layers must be non-negative.')
+
+        self.max_num_nodes_in_dataset = max_num_nodes_in_dataset
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.hidden_dim = hidden_dim
+        self.num_hidden_layers_phi = num_hidden_layers_phi
+        self.num_hidden_layers_rho = num_hidden_layers_rho
+
+        # Input layers
+        self.phi = [
+            nn.Linear(in_features=self.in_channels,
+                      out_features=self.hidden_dim),
+            nn.ReLU(),
+        ]
+        self.rho = []
+
+        for _ in range(self.num_hidden_layers_phi):
+            self.phi.append(
+                nn.Linear(in_features=self.hidden_dim,
+                          out_features=self.hidden_dim))
+            self.phi.append(nn.ReLU())
+
+        for _ in range(self.num_hidden_layers_rho):
+            self.rho.append(
+                nn.Linear(in_features=self.hidden_dim,
+                          out_features=self.hidden_dim))
+            self.rho.append(nn.ReLU())
+
+        # Output layer
+        self.rho.append(
+            nn.Linear(in_features=self.hidden_dim,
+                      out_features=self.out_channels))
+
+        self.phi = nn.Sequential(*self.phi)
+        self.rho = nn.Sequential(*self.rho)
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        def reset(network):
+            for module in network.children():
+                if hasattr(module, 'reset_parameters'):
+                    module.reset_parameters()
+
+        reset(self.phi)
+        reset(self.rho)
+
+    def forward(self, x: Tensor, index: Optional[Tensor] = None,
+                ptr: Optional[Tensor] = None, dim_size: Optional[int] = None,
+                dim: int = -2) -> Tensor:
+        x = self.phi(x)
+        x, _ = to_dense_batch(x, index, fill_value=0,
+                              max_num_nodes=self.max_num_nodes_in_dataset)
```

Comment:
does this functions really need the `self.max_num_nodes_in_dataset` argument. It looks like it'll work fine without that.

Reply:
This is a good observation, the argument should not be necessary for Deep Sets.